### DISCUSSION TITLE
Just clicking on a database in the database navigator immediately re-associates the current scirpt

### DISCUSSION THREAD

**Discussion by rvwoens** [dbeaver/dbeaver#12046] [Q&A]
#### System information: 
- MacOS
- 21.0.2.202104042040 and older

#### Connection specification:
- multiple

#### Describe the problem you're observing:
Just clicking on a database in the database navigator immediately re-associates the current script in the current tab with this database. This re-association seems to be permanent because when closing the tab and reopening the script, the wrong database is still associated with the script.

Script associations should be permanent for a script, as the sql query will (most of the time) not be valid for other databases (other dialect, other tables defined) so a re-association should only be done when you explicitly want this (by right-clicking on the script and using "associate with data source") and not when you are just browing through the datasources in order to open a new tab.

This bug is very annoying. I have about 100 scripts for more than 10 different datasources (mysql, bigquery, progress etc) and every time I run a script I get errors because the script has been silently re-associated.

#### Steps to reproduce, if exist:
- open a script with the correct association
- browse trough the database navigator and open a datasource just to browse its tables
- now the script is permanently re-associated to the (wrong) datasource you are browsing through, completely destroying the script settings.

**Comment by uslss** [comment]
@rvwoens please, uncheck Auto-sync connection with navigator option in main menu
![image](https://user-images.githubusercontent.com/31996417/114006125-d6a39e80-9868-11eb-966b-57d7caee69f7.png)

**Comment by rvwoens** [comment]
Thats it! Have looked for hours in preferences but did not find this. Why this option is default ON (and even exists) is really surprising, as nobody wants this ON in real life situations.

**Reply by p91paul** [reply]
You'd be suprised. While I keep it off as well, I've seen many people asking for this here.
Nobody is a very strong word 😅
